File: Bronze/2015/January/03_its_all_about_the_base/ItsAllAboutTheBase.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.PrintWriter;
import java.util.StringTokenizer;

/**
 * @author Red
 * @version 1.0
 */
public class ItsAllAboutTheBase {
    static int k;
    static char[] baseX;
    static char[] baseY;
    public static void main(String[] args) throws Exception {
        BufferedReader r = new BufferedReader(new FileReader("whatbase.in"));
        PrintWriter pw = new PrintWriter("whatbase.out");
        k = Integer.parseInt(r.readLine());
        for(int i = 0; i < k; i++){
            StringTokenizer st = new StringTokenizer(r.readLine());
            baseX = st.nextToken().toCharArray();
            baseY = st.nextToken().toCharArray();
            int xBase = 10, yBase = 10;
            while(true){
                int x = getNumOnBase(xBase, baseX);
                int y = getNumOnBase(yBase, baseY);
                if(x == y){
                    pw.println(xBase + " " + yBase);
                    break;
                }else if(x < y){
                    xBase++;
                }else{
                    yBase++;
                }
            }
        }
        pw.close();
    }

    /**
     * 计算数字在指定进制下的值
     */
    static int getNumOnBase(int base, char[] num){
        int ans = 0;
        int basePow = 1;
        for(int i = 0 ;i<3;i++){
            ans += (num[2-i]-'0')*basePow;
            basePow *= base;
        }
        return ans;
    }
}

File: Bronze/2015/January/03_its_all_about_the_base/its_all_about_the_base.cpp
#include <iostream>
#include <cstdio>

using namespace std;

#define MAXB 15000

// 计算数字在指定进制下的值
int evaluate(const string& num, int base) {
    return (num[0] - '0') * base * base +
           (num[1] - '0') * base +
           (num[2] - '0');
}

int main() {
    freopen("whatbase.in", "r", stdin);
    freopen("whatbase.out", "w", stdout);

    int T;
    cin >> T;
    for (int t = 1; t <= T; t++) {
        string X, Y;
        cin >> X >> Y;
        // 将两个基数初始化为10，递增得到较小结果的基数，直到它们得到相等的结果。
        int x_base = 10;
        int y_base = 10;
        while (x_base <= MAXB && y_base <= MAXB) {
            int num_x = evaluate(X, x_base);
            int num_y = evaluate(Y, y_base);
            if (num_x < num_y) {
                x_base++;
            } else if (num_y < num_x) {
                y_base++;
            } else {
                cout << x_base << ' ' << y_base << '\n';
                break;
            }
        }
    }
    return 0;
}